akka-actor/src/main/scala/akka/dispatch/affinity/AffinityPool.scala

```diff
@@ -383,16 +352,67 @@ trait RejectionHandlerFactory {
   def create(): RejectionHandler
 }
 
+trait QueueSelectorFactory {
+  def create(): QueueSelector
+}
+
+/**
+ * A `QueueSelector` is responsible for, given a `Runnable` and the number of available
+ * queues, return which of the queues that `Runnable` should be placed in.
+ */
+trait QueueSelector {
+  /**
+   * Must be deterministic—return the same value for the same input.
```

Comment:
I think that maybe the "deterministic" part is a bit of an overstatement. People are free I guess to implement any kind of affinity with this queue selector trait (which is quite great). I am saying all of that because I have been thinking lately about the following scenario. Imagine that someone for some reason might want to have a pool where actors can be scheduled on different number of possible threads based on how long the processing of a message takes. That processing time might actually change over the run of an application and along with it, the output of the queue selector.